### DISCUSSION TITLE
아무 토론 대잔치

### DISCUSSION THREAD

**Discussion by suan073** [suan073/NEEDle_ideaBank#10] [General]
**이곳은 본인의 아이디어나 제안 등을 토론하거나 본인이 배운 지식등을 공유하는 공간입니다.**
1. 번뜩이는 아이디어
2. 개선을 위한 제안
3. 공유하면 좋을 지식들
등 프로젝트를 위한 모든 것을 토론해보는 장입니다.

**Comment by kimmokalover** [comment]
- q1 : Skip-Gram 이란?
- q2: Negative Sampling 이란?

**Reply by kimmokalover** [reply]
**승준**
- **Skip - gram** 은 **중심 단어를 기준으로 corpus 내에 있는 다른 단어들과 연관성**을 구하는 모델이다. 입력층 에서는 중심 단어가 one-hot 백터의 형태로 들어가서 초기에 랜덤한 값으로 초기화된 임베딩 행렬(w1 matrix)와 내적을 한 후 임배딩 백터의 형식으로 분산 표현이 된다.
출력층에서 다시 출력 차원을 맞춰주기 위해 w2 를 내적한 후 output으로 나온 백터와 레이블 백터의 오차를 구한 후 back-propagation을 통해 모든 임베딩 백터의 값을 수정시킨다.

- **Negative Sampling**: 기존의 모델 같은 경우 모든 단어의 임베딩 백터의 값을 수정 해야 했다면, 네거티브 샘플링은 **중심 단어를 기준으로 주변 단어와 주변 단어가 아닌 두 가지 분류**로 나눈다. 이렇게 하면 **이진 분류 문제**가 되기 때문에, 기존의 모델보다 **연산량에 있어 이점을 같는다**.

**Reply by ghost** [reply]
**태민**

- **Skip-gram**은 Word2Vector 모델 중 한가지 이자,  center word(입력/중심 단어)을 기준으로 주변 window-size(양 옆 길이)의 해당하는 context word(출력/주변 단어), 중심단어를 기준으로 주변단어를 구하는 모델이다. 모델은 center word를 input(입력값)으로받아 one-hot vector(size: v, vocab size)의 형식으로 바꾸어 크기가 v * embedding_dim(임베딩 차원)인 랜덤값으로 초기화 된 벡터와 내적을 한다. 내적된 값을 임베딩 벡터라고 한다. 이 임베딩 벡터를 다시 차원에 맞게 내적하여 output(출력값)을 정답 벡터와 loss값을 구하여, loss값을 줄이는 방향으로 임베딩 벡터 값을 수정시킨다.
- **Negative Sampling**은 context word에 틀린 답인 negative word 추가하여 **loss 값을 줄일 수 있다.** loss값이 적게 나오기 때문에, negative sampling이 없는 모델 보다, **연산량이 적어진다.**

**Comment by kimmokalover** [comment]
수인이 해줄일: 기존 .txt 파일에서 조사 제거해주는 프로그램 만들기
참고 : KoNLPy 한국어 처리 패키지[¶](https://datascienceschool.net/03%20machine%20learning/03.01.02%20KoNLPy%20%ED%95%9C%EA%B5%AD%EC%96%B4%20%EC%B2%98%EB%A6%AC%20%ED%8C%A8%ED%82%A4%EC%A7%80.html#konlpy)
